uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -909,8 +919,8 @@ define([
       amendRequestParamsBeforeSave();
 
       return vm.request.isValid()
-        .then(checkIfBalanceChangeHasChanged)
-        .then(decideIfBalanceChangeNeedsAForceRecalculation)
+        .then(isBalanceChangeRecalculationNeeded() && checkIfBalanceChangeHasChanged)
+        .then(isBalanceChangeRecalculationNeededOnBackend)
```

Comment:
Really sorry, but the previous name was better, because `isBalanceChangeRecalculationNeededOnBackend` sounds like it is going to return a boolean value. Please revert this change.